packages/angular/cli/commands/update-impl.ts

```diff
@@ -879,22 +857,63 @@ export class UpdateCommand extends Command<UpdateCommandSchema> {
   }
 
   /**
-   * Checks if the current installed CLI version is older than the latest version.
-   * @returns `true` when the installed version is older.
+   * Checks if the current installed CLI version is older or newer than a compatible version.
+   * @returns the version to install or null when there is no update to install.
    */
-  private async checkCLILatestVersion(verbose = false, next = false): Promise<boolean> {
-    const installedCLIVersion = VERSION.full;
-
-    const LatestCLIManifest = await fetchPackageManifest(
-      `@angular/cli@${next ? 'next' : 'latest'}`,
-      this.logger,
-      {
-        verbose,
-        usingYarn: this.packageManager === PackageManager.Yarn,
-      },
-    );
+  async checkCLIVersion(
+    packagesToUpdate: string[] | undefined,
+    verbose = false,
+    next = false,
+  ): Promise<string | null> {
+    let cliRunnerTargetVersion: string | number;
+    if (next) {
+      cliRunnerTargetVersion = 'next';
+    } else {
+      const updatingAngularPackage = packagesToUpdate?.find((r) => ANGULAR_PACKAGES_REGEXP.test(r));
+      if (updatingAngularPackage) {
+        // If we are updating any Angular package we can update the CLI to the target version because
+        // migrations for @angular/core@13 can be executed using Angular/cli@13.
+        // This is same behaviour as `npx @angular/cli@13 update @angular/core@13`.
+
+        // `@angular/cli@13` -> ['', 'angular/cli', '13']
+        // `@angular/cli` -> ['', 'angular/cli']
+        const tempVersion = coerceVersionNumber(updatingAngularPackage.split('@')[2]);
+        cliRunnerTargetVersion = semver.parse(tempVersion)?.major || 'latest';
+      } else {
+        // When not updating an Angular package we cannot determine which schematic runtime the migration should to be executed in.
+        // Typically, we can assume that the `@angular/cli` was updated previously.
+        // Example: Angular official packages are typically updated prior to NGRX etc...
+        // Therefore, we only update to the latest patch version of the installed major version of the Angular CLI.
+
+        // This is important because we might end up in a scenario where locally Angular v12 is installed, updating NGRX from 11 to 12.
+        // We end up using Angular ClI v13 to run the migrations if we run the migrations using the CLI installed major version + 1 logic.
+        cliRunnerTargetVersion = VERSION.major;
+      }
+    }
 
-    return semver.lt(installedCLIVersion, LatestCLIManifest.version);
+    try {
+      const { version } = await fetchPackageManifest(
+        `@angular/cli@${cliRunnerTargetVersion}`,
+        this.logger,
+        {
+          verbose,
+          usingYarn: this.packageManager === PackageManager.Yarn,
+        },
+      );
+
+      return VERSION.full === version ? null : version;
+    } catch (error) {
+      if (
+        error instanceof Error &&
+        error.message.includes('No matching version found for @angular/cli')
```

Comment:
Which tests fail because of this?  Would it be possible to adjust the tests so that this CI specific error cause isn't needed?  Or are there cases outside of CI where this could happen?

Reply:
Pretty much all the update tests failed, because with verdaccio `@angular/cli@13` will cause the above error. 

I wonder if it will work if we a caret in the version.

Reply:
Ah, that's probably because the local version is `@angular/cli@13.1.0-next.2` and prereleases won't be picked up by `13`.
Maybe we could publish a stable version of 13 (or whatever the current major is) in the test setup so that the local registry can resolve the major.

Reply:
Adjusted the test to remove the need for this try/catch block.